src/features/contracts/contracts-directory.ts

```diff
@@ -177,13 +178,34 @@ export const getContracts = async (type: string): Promise<Contract[]> => {
 		return [];
 	}
 
-	const contractFiles = await glob(
+	let contractFiles = await glob(
 		`${CONTRACTS_BASE_DIR}/**/contracts/${type}/**/*.json`,
 	);
 	if (!contractFiles.length) {
 		return [];
 	}
 
+	// If there are explicit includes, then everything else is excluded so we need to
+	// filter the contractFiles list to include only contracts that are in the INCLUDED_DEVICETYPE_SLUGS map
+	if (INCLUDED_DEVICETYPE_SLUGS.size > 0) {
+		const slugRegex = new RegExp(`/contracts/${type}/([^/]+)/`);
```

Comment:
It'd probably be good to escape `type` to avoid any potential matching issues
```suggestion
		const slugRegex = new RegExp(`/contracts/${_.escapeRegExp(type)}/([^/]+)/`);
```
I know that in practice it probably won't make a difference but I'd rather play it safe